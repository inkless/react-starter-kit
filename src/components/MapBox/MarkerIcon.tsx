import styled, { StyledProps } from 'styled-components';
import { color, ColorProps } from 'styled-system';
import { ReactComponent as MarkerIcon } from '@src/assets/svg/marker.svg';

export type Props = ColorProps;
type ThemeProps = StyledProps<Props>;

export const StyledMarkerIcon = styled(MarkerIcon)`
  position: absolute;
  left: -14px; /* half of the size */
  top: -35px; /* move the icon to top */
  color: ${(p: ThemeProps) => p.theme.colors.primary.main};
  cursor: pointer;
  ${color}
`;

export default StyledMarkerIcon;
