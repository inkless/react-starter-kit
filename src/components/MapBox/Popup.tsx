import React from 'react';
import { Popup } from 'react-map-gl';

import { theme } from '@src/theme';
import Text from '@src/components/base/Typography/Text';

export type PopupInfo = {
  longitude: number;
  latitude: number;
  title: string;
  subtitle: string;
};

export type Props = {
  popupInfo: PopupInfo | null;
  onClose?: () => void;
};

/**
 * Render a Popup in Map
 * When props.popup is null, return null
 *
 * @param props.popupInfo
 * @param props.onClose
 */
export function MapPopup({ popupInfo, onClose }: Props) {
  if (!popupInfo) {
    return null;
  }
  return (
    <Popup
      tipSize={5}
      anchor="top"
      longitude={popupInfo.longitude}
      latitude={popupInfo.latitude}
      closeOnClick={false}
      onClose={onClose}
    >
      <>
        <Text fontSize={3} color={theme.colors.grays[9]} fontWeight="bold" textAlign="center">
          {popupInfo.title}
        </Text>
        <Text fontSize={3} color={theme.colors.grays[7]} p={0}>
          {popupInfo.subtitle}
        </Text>
      </>
    </Popup>
  );
}

export default MapPopup;
