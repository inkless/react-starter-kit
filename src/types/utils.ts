export interface StringTMap<T> {
  [key: string]: T;
}
export interface NumberTMap<T> {
  [key: number]: T;
}

export interface StringAnyMap extends StringTMap<any> {}
export interface NumberAnyMap extends NumberTMap<any> {}

export interface StringStringMap extends StringTMap<string> {}
export interface NumberStringMap extends NumberTMap<string> {}

export interface StringNumberMap extends StringTMap<number> {}
export interface NumberNumberMap extends NumberTMap<number> {}

export interface StringBooleanMap extends StringTMap<boolean> {}
export interface NumberBooleanMap extends NumberTMap<boolean> {}

export type valueof<T> = T[keyof T];

/**
 * Omit some properties in one type.
 *
 * e.g.
 * ```
 * type Foo = { apple: string; banana: string };
 * type Bar = Omit<Foo, 'apple'>;
 * ```
 * so `Bar` is equivalant to `{ banana: string }`
 */
export type Omit<T, K> = Pick<T, Exclude<keyof T, K>>;
