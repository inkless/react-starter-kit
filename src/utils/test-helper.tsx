import React from 'react';
import { mount, render, shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import { theme } from '@src/theme';
import { createRootReducer } from '@src/store/reducers';

export function renderWithTheme(tree: JSX.Element, options?: any) {
  return render(tree, {
    ...options,
    context: { theme },
  });
}

export function shallowWithTheme(tree: JSX.Element, options?: any) {
  return shallow(tree, {
    ...options,
    context: { theme },
  });
}

export function mountWithTheme(tree: JSX.Element, options?: any) {
  return mount(<ThemeProvider theme={theme}>{tree}</ThemeProvider>, options);
}

export function createInitialState() {
  const rootReducer = createRootReducer({} as any);
  return rootReducer(undefined, '' as any);
}
