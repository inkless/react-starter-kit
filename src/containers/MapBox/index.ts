import { connect } from 'react-redux';

import MapBox from '@src/components/MapBox';
import { RootState } from '@src/store';
import { fetchOriginOperation } from './operations';
import { setViewport } from './actions';
import { mapStyleSelector, getDeliveringOrders } from './selectors';

const mapStateToProps = (state: RootState) => ({
  ...state.mapbox,
  mapStyle: mapStyleSelector(state),
  deliveringOrders: getDeliveringOrders(state),
});

const mapDispatchToProps = {
  fetchOriginOperation,
  setViewport,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(MapBox);
