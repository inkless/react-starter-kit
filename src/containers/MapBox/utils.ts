import request from '@src/utils/request';
import { lineString, Feature } from '@turf/helpers';
import turfLength from '@turf/length';
import { decode } from '@mapbox/polyline';

import { StringAnyMap } from '@src/types/utils';
import { MAPBOX_API_URL } from './constants';

const ACCESS_TOKEN = process.env.MapboxAccessToken;
const POLYLINE_PRECISION = 5;
const SECONDS_IN_HOUR = 3600;
const SPEED_BASE = 20;

/**
 * Fetch driving route from origin to destination
 *
 * @param origin
 * @param destination
 */
export async function fetchDrivingRoute(origin: [number, number], destination: [number, number]) {
  const baseUrl = `${MAPBOX_API_URL}/directions/v5/mapbox/driving-traffic`;
  const coordinates = `${origin.join(',')};${destination.join(',')}`;
  const url = `${baseUrl}/${coordinates}?steps=true&overview=full&access_token=${ACCESS_TOKEN}`;
  const data = await request(url);
  return data.routes[0];
}

/**
 * Given any address, find the geocoding
 * @param address
 */
export async function fetchGeocoding(address: string) {
  const url = `${MAPBOX_API_URL}/geocoding/v5/mapbox.places/${encodeURIComponent(
    address
  )}.json?access_token=${ACCESS_TOKEN}`;
  const data = await request(url);
  return data.features[0].center;
}

/**
 * Generate a speed between 20 to 25 miles per hour
 * @returns miles per second
 */
export function generateRandomSpeed() {
  return (Math.random() * 5 + SPEED_BASE) / SECONDS_IN_HOUR;
}

/**
 * Decode geometry and return geojson route
 * @param geometry
 * @returns geojson lineString
 */
export function createRouteFromGeometry(geometry: string, options?: StringAnyMap) {
  const decoded = decode(geometry, POLYLINE_PRECISION).map(c => {
    return c.reverse();
  });

  return lineString(decoded, options);
}

/**
 * Given a route, calculate the distance of it
 * @param route
 * @returns distance in miles
 */
export function calculateRouteLength(route: Feature) {
  return turfLength(route, { units: 'miles' });
}
