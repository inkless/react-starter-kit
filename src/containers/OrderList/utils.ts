import { groupBy } from '@src/utils/array';
import { PendingOrder } from './types';

/**
 * reduce all orders' dispatchTime by 1 second,
 * move them to delivering if dispatchTime is 0
 *
 * @param orders - pending orders
 * @returns
 */
export function tickOneSecond(orders: PendingOrder[]) {
  const updatedOrders = orders.map(order => ({
    ...order,
    dispatchTime: order.dispatchTime - 1,
  }));

  const { true: delivering = [], false: pending = [] } = groupBy(
    updatedOrders,
    ({ dispatchTime }) => dispatchTime === 0
  );

  return {
    delivering,
    pending,
  };
}
