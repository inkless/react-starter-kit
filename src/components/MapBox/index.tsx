import React, { useEffect, useState } from 'react';
import ReactMapGL, { Marker, ViewState } from 'react-map-gl';

import { theme } from '@src/theme';
import Flex, { Props as FlexProps } from '@src/components/base/Grid/Flex';
import { DeliveringOrder } from '@src/containers/OrderList/types';
import { MapboxState } from '@src/containers/MapBox/types';
import MarkerIcon from './MarkerIcon';
import Popup, { PopupInfo } from './Popup';

export type Props = {
  origin: MapboxState['origin'];
  viewport: ViewState;
  deliveringOrders: DeliveringOrder[];
  mapStyle: any;
  deliverSimulationStarted: boolean;

  // actions
  setViewport: (v: ViewState) => void;
  // operations
  fetchOriginOperation: () => void;
} & FlexProps;

/**
 * Render the MapBox container
 */
export function MapBox({ deliveringOrders, color, ...props }: Props) {
  const {
    coordinates: [originLongitude, originLatitude],
    address,
  } = props.origin;

  // It's not really needed to move this state to redux, so useState instead
  const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
  const handleMarkerClick = (info: PopupInfo | null) => {
    return () => setPopupInfo(info);
  };

  // fetchOriginOperation when component is mounted
  useEffect(() => {
    if (!props.deliverSimulationStarted) {
      props.fetchOriginOperation();
    }
  }, [] /* only execute once */);

  return (
    <Flex {...props}>
      <ReactMapGL
        mapStyle={props.mapStyle}
        {...props.viewport}
        width="100%"
        height="100%"
        onViewportChange={props.setViewport}
      >
        {/* Render Home Marker */}
        <Marker longitude={originLongitude} latitude={originLatitude}>
          <MarkerIcon
            color={theme.colors.common.red}
            onClick={handleMarkerClick({
              longitude: originLongitude,
              latitude: originLatitude,
              title: 'Home',
              subtitle: address,
            })}
          />
        </Marker>

        {/* Render Delivering Markers */}
        {deliveringOrders.map(
          ({ coordinates: [longitude, latitude], destination, name }, index) => (
            <Marker key={index} longitude={longitude} latitude={latitude}>
              <MarkerIcon
                onClick={handleMarkerClick({
                  longitude,
                  latitude,
                  subtitle: destination,
                  title: name,
                })}
              />
            </Marker>
          )
        )}

        {/* Render Popup */}
        <Popup popupInfo={popupInfo} onClose={handleMarkerClick(null)} />
      </ReactMapGL>
    </Flex>
  );
}

MapBox.defaultProps = {
  padding: 0,
};

export default MapBox;
