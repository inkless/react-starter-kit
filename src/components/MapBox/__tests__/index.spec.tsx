import React from 'react';

import { props as mapProps } from './__fixture__/mapbox';
import MapBox, { Props as MapBoxProps } from '..';
import { mountWithTheme } from '@src/utils/test-helper';

describe('Component MapBox', () => {
  let props: MapBoxProps;
  beforeEach(() => {
    props = {
      ...mapProps,
    };
  });

  afterAll(() => {
    (props as any) = null;
  });

  it('should render MapBox correctly', () => {
    // shallow does not really support useEffect yet
    // so using mountWithTheme instead
    // https://github.com/airbnb/enzyme/issues/2011
    const component = mountWithTheme(<MapBox {...props} />);

    expect(props.fetchOriginOperation).toHaveBeenCalledTimes(1);
    expect(props.setViewport).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});
