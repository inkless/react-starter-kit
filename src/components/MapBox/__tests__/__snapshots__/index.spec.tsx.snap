// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component MapBox should render MapBox correctly 1`] = `
.c0 {
  box-sizing: border-box;
  padding: 8px;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<ThemeProvider
  theme={
    Object {
      "Badge": Object {},
      "Box": Object {
        "padding": "8px",
      },
      "Button": Object {},
      "Card": Object {},
      "Footer": Object {},
      "Heading": Object {},
      "Text": Object {},
      "colors": Object {
        "common": Object {
          "black": "#000",
          "blue": "#2196f3",
          "red": "#fd5850",
          "white": "#fff",
        },
        "disabled": Object {
          "contrast": "#ffffff",
          "main": "#b1b1b1",
        },
        "error": Object {
          "contrast": "#ffffff",
          "main": "#d46e6e",
        },
        "grays": Array [
          "#f2f2f2",
          "#e4e4e4",
          "#d4d4d4",
          "#c3c3c3",
          "#b1b1b1",
          "#9b9b9b",
          "#8b8b8b",
          "#7a7a7a",
          "#656565",
          "#484848",
        ],
        "hover": Object {
          "main": "rgba(0, 0, 0, 0.08)",
        },
        "primary": Object {
          "contrast": "#ffffff",
          "dark": "#3b8156",
          "light": "#9be3b3",
          "main": "#6ab183",
        },
        "secondary": Object {
          "contrast": "#ffffff",
          "dark": "#0a151d",
          "light": "#5b656f",
          "main": "#313b44",
        },
      },
      "fontSizes": Array [
        8,
        10,
        12,
        14,
        16,
        20,
        24,
        32,
        48,
        64,
        96,
        128,
      ],
      "fontWeights": Object {
        "bold": 700,
        "normal": 500,
      },
      "fonts": Object {
        "font": "-apple-system,Helvetica,Arial,sans-serif",
        "mono": "SFMono-Regular,Menlo,monospace",
      },
      "letterSpacings": Object {
        "caps": "0.25em",
        "normal": "normal",
      },
      "radii": Array [
        0,
        2,
        4,
        8,
      ],
      "shadows": Array [
        "none",
        "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
        "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
        "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
        "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
        "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
        "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
        "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
        "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
        "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
        "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
        "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
        "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
        "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
        "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
        "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
        "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
        "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
        "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        32,
        64,
        128,
        256,
        512,
      ],
      "transitions": Object {
        "create": [Function],
        "duration": Object {
          "complex": 375,
          "enteringScreen": 225,
          "leavingScreen": 195,
          "short": 250,
          "shorter": 200,
          "shortest": 150,
          "standard": 300,
        },
        "easing": Object {
          "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
          "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
          "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
          "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
        },
      },
    }
  }
>
  <MapBox
    deliverSimulationStarted={false}
    deliveringOrders={
      Array [
        Object {
          "coordinates": Array [
            -131,
            37,
          ],
          "destination": "185 Culver Blvd, Playa Del Rey, CA 90293",
          "dispatchTime": 119,
          "distance": 2,
          "id": "9H4whD",
          "name": "Banana Split",
          "route": Object {},
          "speed": 22,
          "startTime": 2019-04-20T05:01:15.265Z,
        },
      ]
    }
    fetchOriginOperation={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    mapStyle={Object {}}
    origin={
      Object {
        "address": "mock address",
        "coordinates": Array [
          -130,
          38,
        ],
      }
    }
    padding={0}
    setViewport={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "altitude": 1.5,
              "bearing": 0,
              "height": 0,
              "latitude": 37,
              "longitude": -130,
              "maxPitch": 60,
              "maxZoom": 24,
              "minPitch": 0,
              "minZoom": 0,
              "pitch": 0,
              "transitionDuration": 0,
              "transitionEasing": [Function],
              "transitionInterpolator": LinearInterpolator {
                "around": undefined,
                "propNames": Array [
                  "longitude",
                  "latitude",
                  "zoom",
                  "bearing",
                  "pitch",
                ],
              },
              "transitionInterruption": 1,
              "width": 0,
              "zoom": 11,
            },
            Object {},
            Object {},
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    viewport={
      Object {
        "latitude": 37,
        "longitude": -130,
        "zoom": 11,
      }
    }
  >
    <Styled(styled.div)
      deliverSimulationStarted={false}
      fetchOriginOperation={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      mapStyle={Object {}}
      origin={
        Object {
          "address": "mock address",
          "coordinates": Array [
            -130,
            38,
          ],
        }
      }
      padding={0}
      setViewport={
        [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "altitude": 1.5,
                "bearing": 0,
                "height": 0,
                "latitude": 37,
                "longitude": -130,
                "maxPitch": 60,
                "maxZoom": 24,
                "minPitch": 0,
                "minZoom": 0,
                "pitch": 0,
                "transitionDuration": 0,
                "transitionEasing": [Function],
                "transitionInterpolator": LinearInterpolator {
                  "around": undefined,
                  "propNames": Array [
                    "longitude",
                    "latitude",
                    "zoom",
                    "bearing",
                    "pitch",
                  ],
                },
                "transitionInterruption": 1,
                "width": 0,
                "zoom": 11,
              },
              Object {},
              Object {},
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
      viewport={
        Object {
          "latitude": 37,
          "longitude": -130,
          "zoom": 11,
        }
      }
    >
      <StyledComponent
        deliverSimulationStarted={false}
        fetchOriginOperation={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": false,
              "lastClassName": "c0",
              "rules": Array [
                "
  box-sizing: border-box;
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
",
                "
  display: flex;
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
  ",
                [Function],
                "
",
              ],
            },
            "displayName": "Styled(styled.div)",
            "foldedComponentIds": Array [
              "sc-bdVaJa",
            ],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        mapStyle={Object {}}
        origin={
          Object {
            "address": "mock address",
            "coordinates": Array [
              -130,
              38,
            ],
          }
        }
        padding={0}
        setViewport={
          [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "altitude": 1.5,
                  "bearing": 0,
                  "height": 0,
                  "latitude": 37,
                  "longitude": -130,
                  "maxPitch": 60,
                  "maxZoom": 24,
                  "minPitch": 0,
                  "minZoom": 0,
                  "pitch": 0,
                  "transitionDuration": 0,
                  "transitionEasing": [Function],
                  "transitionInterpolator": LinearInterpolator {
                    "around": undefined,
                    "propNames": Array [
                      "longitude",
                      "latitude",
                      "zoom",
                      "bearing",
                      "pitch",
                    ],
                  },
                  "transitionInterruption": 1,
                  "width": 0,
                  "zoom": 11,
                },
                Object {},
                Object {},
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
        viewport={
          Object {
            "latitude": 37,
            "longitude": -130,
            "zoom": 11,
          }
        }
      >
        <div
          className="c0"
          origin={
            Object {
              "address": "mock address",
              "coordinates": Array [
                -130,
                38,
              ],
            }
          }
        >
          <InteractiveMap
            asyncRender={false}
            attributionControl={true}
            bearing={0}
            className=""
            clickRadius={0}
            container={<body />}
            disableTokenWarning={false}
            doubleClickZoom={true}
            dragPan={true}
            dragRotate={true}
            getCursor={[Function]}
            height="100%"
            keyboard={true}
            latitude={37}
            longitude={-130}
            mapOptions={Object {}}
            mapStyle={Object {}}
            mapboxApiAccessToken="no-token"
            maxPitch={60}
            maxZoom={24}
            minPitch={0}
            minZoom={0}
            onClick={null}
            onContextMenu={[Function]}
            onError={[Function]}
            onHover={null}
            onLoad={[Function]}
            onNativeClick={null}
            onResize={[Function]}
            onStateChange={[Function]}
            onTransitionEnd={[Function]}
            onTransitionInterrupt={[Function]}
            onTransitionStart={[Function]}
            onViewStateChange={null}
            onViewportChange={
              [MockFunction] {
                "calls": Array [
                  Array [
                    Object {
                      "altitude": 1.5,
                      "bearing": 0,
                      "height": 0,
                      "latitude": 37,
                      "longitude": -130,
                      "maxPitch": 60,
                      "maxZoom": 24,
                      "minPitch": 0,
                      "minZoom": 0,
                      "pitch": 0,
                      "transitionDuration": 0,
                      "transitionEasing": [Function],
                      "transitionInterpolator": LinearInterpolator {
                        "around": undefined,
                        "propNames": Array [
                          "longitude",
                          "latitude",
                          "zoom",
                          "bearing",
                          "pitch",
                        ],
                      },
                      "transitionInterruption": 1,
                      "width": 0,
                      "zoom": 11,
                    },
                    Object {},
                    Object {},
                  ],
                ],
                "results": Array [
                  Object {
                    "type": "return",
                    "value": undefined,
                  },
                ],
              }
            }
            pitch={0}
            preserveDrawingBuffer={false}
            preventStyleDiffing={false}
            reuseMaps={false}
            scrollZoom={true}
            style={null}
            touchAction="none"
            touchRotate={false}
            touchZoom={true}
            transitionDuration={0}
            transitionEasing={[Function]}
            transitionInterpolator={
              LinearInterpolator {
                "around": undefined,
                "propNames": Array [
                  "longitude",
                  "latitude",
                  "zoom",
                  "bearing",
                  "pitch",
                ],
              }
            }
            transitionInterruption={1}
            visibilityConstraints={
              Object {
                "maxPitch": 60,
                "maxZoom": 24,
                "minPitch": 0,
                "minZoom": 0,
              }
            }
            visible={true}
            width="100%"
            zoom={11}
          >
            <div
              key="event-canvas"
              style={
                Object {
                  "cursor": "grab",
                  "height": "100%",
                  "position": "relative",
                  "width": "100%",
                }
              }
            >
              <StaticMap
                asyncRender={false}
                attributionControl={true}
                bearing={0}
                className=""
                clickRadius={0}
                container={<body />}
                disableTokenWarning={false}
                doubleClickZoom={true}
                dragPan={true}
                dragRotate={true}
                getCursor={[Function]}
                height="100%"
                keyboard={true}
                latitude={37}
                longitude={-130}
                mapOptions={Object {}}
                mapStyle={Object {}}
                mapboxApiAccessToken="no-token"
                maxPitch={60}
                maxZoom={24}
                minPitch={0}
                minZoom={0}
                onClick={null}
                onContextMenu={[Function]}
                onError={[Function]}
                onHover={null}
                onLoad={[Function]}
                onNativeClick={null}
                onResize={[Function]}
                onStateChange={[Function]}
                onTransitionEnd={[Function]}
                onTransitionInterrupt={[Function]}
                onTransitionStart={[Function]}
                onViewStateChange={null}
                onViewportChange={
                  [MockFunction] {
                    "calls": Array [
                      Array [
                        Object {
                          "altitude": 1.5,
                          "bearing": 0,
                          "height": 0,
                          "latitude": 37,
                          "longitude": -130,
                          "maxPitch": 60,
                          "maxZoom": 24,
                          "minPitch": 0,
                          "minZoom": 0,
                          "pitch": 0,
                          "transitionDuration": 0,
                          "transitionEasing": [Function],
                          "transitionInterpolator": LinearInterpolator {
                            "around": undefined,
                            "propNames": Array [
                              "longitude",
                              "latitude",
                              "zoom",
                              "bearing",
                              "pitch",
                            ],
                          },
                          "transitionInterruption": 1,
                          "width": 0,
                          "zoom": 11,
                        },
                        Object {},
                        Object {},
                      ],
                    ],
                    "results": Array [
                      Object {
                        "type": "return",
                        "value": undefined,
                      },
                    ],
                  }
                }
                pitch={0}
                preserveDrawingBuffer={false}
                preventStyleDiffing={false}
                reuseMaps={false}
                scrollZoom={true}
                style={null}
                touchAction="none"
                touchRotate={false}
                touchZoom={true}
                transitionDuration={0}
                transitionEasing={[Function]}
                transitionInterpolator={
                  LinearInterpolator {
                    "around": undefined,
                    "propNames": Array [
                      "longitude",
                      "latitude",
                      "zoom",
                      "bearing",
                      "pitch",
                    ],
                  }
                }
                transitionInterruption={1}
                visibilityConstraints={
                  Object {
                    "maxPitch": 60,
                    "maxZoom": 24,
                    "minPitch": 0,
                    "minZoom": 0,
                  }
                }
                visible={true}
                width="100%"
                zoom={11}
              >
                <div
                  key="map-container"
                  style={
                    Object {
                      "height": "100%",
                      "position": "relative",
                      "width": "100%",
                    }
                  }
                >
                  <div
                    className=""
                    key="map-mapbox"
                    style={
                      Object {
                        "height": "100%",
                        "overflow": "hidden",
                        "position": "absolute",
                        "visibility": "inherit",
                        "width": "100%",
                      }
                    }
                  />
                  <AutoSizer
                    disableHeight={false}
                    disableWidth={false}
                    key="autosizer"
                    onResize={[Function]}
                    style={Object {}}
                  >
                    <div
                      style={
                        Object {
                          "height": 0,
                          "overflow": "visible",
                          "width": 0,
                        }
                      }
                    />
                  </AutoSizer>
                </div>
              </StaticMap>
            </div>
          </InteractiveMap>
        </div>
      </StyledComponent>
    </Styled(styled.div)>
  </MapBox>
</ThemeProvider>
`;
